core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.ts

```diff
@@ -18,105 +18,121 @@
+                explorationId, exploration);
+
+              if (successCallback) {
+                successCallback(exploration);
+              }
+            }, function(errorResponse) {
+              if (errorCallback) {
+                errorCallback(errorResponse.data);
+              }
+            }
+          );
         }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
+      );
     };
 
     var _deleteExploration = function(
         explorationId, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(explorationId, null);
-
-      $http['delete'](editableExplorationDataUrl).then(function() {
-        // Delete item from the ReadOnlyExplorationBackendApiService's cache
-        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
-          explorationId);
-        if (successCallback) {
-          successCallback({});
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
+      _getExplorationUrlAsync(explorationId, null).then(
+        function(editableExplorationDataUrl) {
+          $http['delete'](editableExplorationDataUrl).then(function() {
+            // Delete item from the ReadOnlyExplorationBackendApiService's cache
+            ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+              explorationId);
+            if (successCallback) {
+              successCallback({});
+            }
+          }, function(errorResponse) {
+            if (errorCallback) {
+              errorCallback(errorResponse.data);
+            }
+          });
         }
-      });
+      );
     };
 
-    var _getExplorationUrl = function(explorationId, applyDraft) {
+    var _getExplorationUrlAsync = function(explorationId, applyDraft) {
       if (applyDraft) {
-        return UrlInterpolationService.interpolateUrl(
+        return $q.resolve(UrlInterpolationService.interpolateUrl(
           EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE, {
             exploration_id: explorationId,
             apply_draft: JSON.stringify(applyDraft)
-          });
-      }
-      if (!GLOBALS.can_edit && GLOBALS.can_voiceover) {
-        return UrlInterpolationService.interpolateUrl(
-          VOICEOVER_EXPLORATION_DATA_URL_TEMPLATE, {
-            exploration_id: explorationId
-          });
+          })
+        );
       }
-
-      return UrlInterpolationService.interpolateUrl(
-        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
-          exploration_id: explorationId
-        });
+      return UserExplorationPermissionsService.getPermissionsAsync().then(
+        function(permissions) {
+          if (!permissions.can_edit && permissions.can_voiceover) {
+            return UrlInterpolationService.interpolateUrl(
+              VOICEOVER_EXPLORATION_DATA_URL_TEMPLATE, {
```

Comment:
As per discussion, we were thinking of making one url for voice artists and editor, right? We aren't resolving it in this PR?

Reply:
I gave it some thought and I think it would be better to do it in a different PR in order to reduce the scope of this PR

Reply:
I think it would be better to do it in this PR actually. Am I correct that if you just had one URL then this wouldn't need to be async? There is significant benefit in not having things be async, so it might be better to do that PR first and then this one (if you want to keep them separate).

Reply:
> I think it would be better to do it in this PR actually

Done